lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -772,16 +795,19 @@ class TypeRefinementContextBuilder : private ASTWalker {
       FalseRefinement = FalseFlow;
     }
 
-    if (NestedCount == 0)
-      return std::make_pair(None, FalseRefinement);
-
```

Comment:
Q: Why is this branch being removed?

Reply:
I see what you mean. The reason I removed it is just because I didn't want to duplicate the logic that inverts the result. It's not clear if there's a consequence of returning the current refinement instead of None, but I added it back with a closure instead.